runtime/calamari/src/weights/pallet_manta_pay.rs

```diff
@@ -45,27 +45,55 @@ use sp_std::marker::PhantomData;
 
 /// Weight functions needed for pallet_manta_pay.
 pub trait WeightInfo {
-    fn to_private() -> Weight;
-    fn to_public() -> Weight;
-    fn private_transfer() -> Weight;
-    fn public_transfer() -> Weight;
+    fn to_private_native() -> Weight;
+    fn to_private_non_native() -> Weight;
+    fn to_public_native() -> Weight;
+    fn to_public_non_native() -> Weight;
+    fn private_transfer_native() -> Weight;
+    fn private_transfer_non_native() -> Weight;
+    fn public_transfer_native() -> Weight;
+    fn public_transfer_non_native() -> Weight;
 }
 
 /// Weights for pallet_manta_pay using the Substrate node and recommended hardware.
 pub struct SubstrateWeight<T>(PhantomData<T>);
 impl<T: frame_system::Config> pallet_manta_pay::WeightInfo for SubstrateWeight<T> {
+    // Storage: MantaPay UtxoSet (r:1 w:1)
+    // Storage: MantaPay NullifierSetSize (r:1 w:0)
+    // Storage: MantaPay ShardTrees (r:1 w:1)
+    // Storage: System Account (r:1 w:1)
+    // Storage: MantaPay UtxoAccumulatorOutputs (r:0 w:1)
+    // Storage: MantaPay Shards (r:0 w:1)
+    fn to_private_native() -> Weight {
+        (26_934_980_000 as Weight)
+            .saturating_add(T::DbWeight::get().reads(4 as Weight))
+            .saturating_add(T::DbWeight::get().writes(5 as Weight))
+    }
     // Storage: Assets Asset (r:1 w:1)
     // Storage: Assets Account (r:2 w:2)
     // Storage: MantaPay UtxoSet (r:1 w:1)
     // Storage: MantaPay NullifierSetSize (r:1 w:0)
     // Storage: MantaPay ShardTrees (r:1 w:1)
     // Storage: MantaPay UtxoAccumulatorOutputs (r:0 w:1)
     // Storage: MantaPay Shards (r:0 w:1)
-    fn to_private() -> Weight {
-        (39_380_718_000 as Weight)
+    fn to_private_non_native() -> Weight {
+        (27_130_516_000 as Weight)
```

Comment:
if native and non-native assets aare just 200k units apart, why even distinguish between them?

Reply:
1. If we wanted to keep this, it could be worthwhile to find out whether one can manually invoke the benchmarking function just to compare the outputs (making sure you compare worst-case execution times) of these two in a unit test and then discard the computed weight result.

2. Alternative would be to make it a simple unit test that runs one native and one non-native asset to_private/transfer/public and fails if their returned weights differ by too much.
This doesn't catch the worst-case execution time of both of these tho.

If 1. turns out to be too complex/not possible I would vote for lower complexity and closing here